lms/djangoapps/mobile_api/course_info/views.py

```diff
@@ -166,3 +170,119 @@ def post(self, request, *args, **kwargs):
         # Populate user activity for tracking progress towards a user's course goals
         UserActivity.record_user_activity(user, course_key)
         return Response(status=(200))
+
+
+@view_auth_classes(is_authenticated=False)
+class BlocksInfoInCourseView(BlocksInCourseView):
+    """
+    **Use Case**
+
+        Returns the blocks in the course according to the requesting user's access level.
+        Add to response info fields with information about course
+
+    **Example requests**:
+
+        This api works with all versions {api_version}, you can use: v0.5, v1, v2 or v3
+
+        GET /api/mobile/{api_version}/course_info/blocks/?course_id=<course_id>
+        GET /api/mobile/{api_version}/course_info/blocks/?course_id=<course_id>
+            &username=anjali
+            &depth=all
+            &requested_fields=graded,format,student_view_multi_device,lti_url
+            &block_counts=video
+            &student_view_data=video
+            &block_types_filter=problem,html
+
+    **Response example**
+
+        Body consists of the following fields:
+
+        root: (str) The ID of the root node of the requested course block structure.\
+        blocks: (dict) A dictionary or list, based on the value of the
+            "return_type" parameter. Maps block usage IDs to a collection of
+            information about each block. Each block contains the following
+            fields.
+
+        id: (str) The Course's id (Course Run key)
```

Comment:
what if we call the view with list param, will we still get these fields?

Reply:
If we call this view with return_type=list we will not get certificate and all other details processed in this view. Please update view according to the desired behavior.

Reply:
@jawad-khan, the docstring was updated to explicitly include data about the response when return_type=list is provided